src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/math/RobustBrentSolver.java

```diff
@@ -0,0 +1,439 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.math;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.math3.analysis.UnivariateFunction;
+import org.apache.commons.math3.analysis.solvers.AbstractUnivariateSolver;
+import org.apache.commons.math3.analysis.solvers.BaseAbstractUnivariateSolver;
+import org.apache.commons.math3.exception.NoBracketingException;
+import org.apache.commons.math3.exception.NumberIsTooLargeException;
+import org.apache.commons.math3.exception.TooManyEvaluationsException;
+import org.apache.commons.math3.util.FastMath;
+import org.apache.commons.math3.util.Precision;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A robust version of the Brent solver that tries to avoid spurious non-bracketing conditions.
```

Comment:
I think we need to at least explicitly attribute the code borrowed from the Apache Commons BrentSolver implementation, but IANAL. @vdauwera can probably comment.